src/murmur/ServerDB.cpp

```diff
@@ -1981,22 +1981,50 @@ int Server::readLastChannel(int id) {
 	if (id < 0)
 		return -1;
 
+	if (!Meta::mp.bRememberChan)
+		return -1;
+
 	TransactionHolder th;
 	QSqlQuery &query = *th.qsqQuery;
 
-	SQLPREP("SELECT `lastchannel` FROM `%1users` WHERE `server_id` = ? AND `user_id` = ?");
+	SQLPREP("SELECT `lastchannel`,`last_seen` FROM `%1users` WHERE `server_id` = ? AND `user_id` = ?");
 	query.addBindValue(iServerNum);
 	query.addBindValue(id);
 	SQLEXEC();
 
 	if (query.next()) {
 		int cid = query.value(0).toInt();
-		if (qhChannels.contains(cid))
-			return cid;
+		QDateTime last_seen = QDateTime::fromString(query.value(1).toString(), Qt::ISODate);
```

Comment:
Here we should handle what happens, if the field is not set. So first check if this is an empty String (or whatever it is, if it isn't set) and then act accordingly (if it's not set I'd say screw it and just act as if the duration didn't pass yet - it's most likely an upgrade issue that only happens once per user)